packages/trader/src/Stores/Modules/Trading/Helpers/chart.ts

```diff
@@ -0,0 +1,233 @@
+    CHOOSE_TIME_INTERVAL: 'choose_time_interval',
+    CLEAN_ALL_ACTIVE: 'clean_all_active',
+    CLOSE: 'close',
+    DELETE_ACTIVE: 'delete_active',
+    DELETE_FROM_FAVORITES: 'delete_from_favorites',
+    EDIT_ACTIVE: 'edit_active',
+    INFO_OPEN: 'info_open',
+    INFO_CLOSE: 'info_close',
+    INFO_REDIRECT: 'info_redirect',
+    OPEN: 'open',
+    SEARCH: 'search',
+} as const;
+
+export const STATE_TYPES = {
+    CHART_INTERVAL_CHANGE: 'CHART_INTERVAL_CHANGE',
+    CHART_MODE_TOGGLE: 'CHART_MODE_TOGGLE',
+    CHART_TYPE_CHANGE: 'CHART_TYPE_CHANGE',
+    FAVORITE_MARKETS_TOGGLE: 'FAVORITE_MARKETS_TOGGLE',
+    INDICATOR_ADDED: 'INDICATOR_ADDED',
+    INDICATOR_DELETED: 'INDICATOR_DELETED',
+    INDICATOR_INFO_TOGGLE: 'INDICATOR_INFO_TOGGLE',
+    INDICATOR_SEARCH: 'INDICATOR_SEARCH',
+    INDICATOR_SETTINGS_OPEN: 'INDICATOR_SETTINGS_OPEN',
+    INDICATORS_CLEAR_ALL: 'INDICATORS_CLEAR_ALL',
+    INDICATORS_MODAL_TOGGLE: 'INDICATORS_MODAL_TOGGLE',
+    INITIAL: 'INITIAL',
+    MARKET_INFO_REDIRECT: 'MARKET_INFO_REDIRECT',
+    MARKET_SEARCH: 'MARKET_SEARCH',
+    MARKET_STATE_CHANGE: 'MARKET_STATE_CHANGE',
+    MARKETS_LIST_TOGGLE: 'MARKETS_LIST_TOGGLE',
+    READY: 'READY',
+    SCROLL_TO_LEFT: 'SCROLL_TO_LEFT',
+    SYMBOL_CHANGE: 'SYMBOL_CHANGE',
+} as const;
+
+export const SUBFORM_NAME = {
+    INDICATORS_INFO: 'indicators_info',
+    INDICATORS_TYPE: 'indicators_type',
+} as const;
+
+const getChartTypeFormAnalyticsData = (state: keyof typeof STATE_TYPES, option: TStateChangeOption) => {
+    const { chart_type_name = '', is_open, time_interval_name } = option;
+    const chart_event_type = 'ce_chart_types_form';
+    const payload = {
+        data: {
+            action: '',
+            chart_type_name,
+            time_interval_name,
+        },
+        event_type: chart_event_type,
+    };
+    const open_close_action = is_open ? ACTION.OPEN : ACTION.CLOSE;
+    if (!chart_type_name) return {};
+    if (state === STATE_TYPES.CHART_MODE_TOGGLE) {
+        payload.data.action = open_close_action;
+        return payload;
+    } else if (state === STATE_TYPES.CHART_TYPE_CHANGE) {
+        payload.data.action = ACTION.CHOOSE_CHART_TYPE;
+        return payload;
+    } else if (state === STATE_TYPES.CHART_INTERVAL_CHANGE) {
+        payload.data.action = ACTION.CHOOSE_TIME_INTERVAL;
+        return payload;
+    }
+    return {};
+};
+
+const getIndicatorTypeFormAnalyticsData = (state: keyof typeof STATE_TYPES, option: TStateChangeOption) => {
+    const { indicator_type_name = '', indicators_category_name = '', is_info_open, is_open, search_string } = option;
+    const indicators_event_type = 'ce_indicators_types_form';
+    const indicators_subform = is_info_open ? SUBFORM_NAME.INDICATORS_INFO : SUBFORM_NAME.INDICATORS_TYPE;
+    const info_open_close_action = is_info_open ? ACTION.INFO_OPEN : ACTION.INFO_CLOSE;
+    const open_close_action = is_open ? ACTION.OPEN : ACTION.CLOSE;
+    if (indicator_type_name) {
```

Comment:
[Suggestion] Could you use Switch case all over this file.. I think it will be more readable.

Reply:
@matin-deriv I used switch-case first but I changed it to if-else because switch-case doesn't suit my purpose, every case is treated in a different way so I'll have to add if-else inside each case anyway, that's why i decided not to use switch-case )